Update collab exercise (#89)
ImperialCollegeLondon/intermediate_grad_school_git_course: _episodes/l2-04-colaborative-development.md

> and you are using git for version control and GitHub to collaborate in the
> writing of the book.
>
> Form groups of 3-4 people and choose one to act as Administrator.
>
> **Housekeeping** (Administrator's task):
>
> - Create a new repository using the template [Book of Recipes repository](https://github.com/ImperialCollegeLondon/book_of_recipes) which has the skeleton of the book. (**Note**: Click on the green  "Use this template" button on the top left of the Book of Recipes repository and then select "Create a new repository")
>
>   ![Use this template]({{ site.baseurl }}/fig/use_this_template.png "Use this template"){:class="img-responsive" style="float: left; margin-right: 10px; width: 20%;"}
>
>   <div style="clear: both;"></div>
>

> > - Set the owner of the new repository as your own GitHub username and the name of the repository the same ("book_of_recipes").
> > - Set the description as "Repository for the exercises of the Grad School Git Course".
> > - Make sure to select the "Public" option for your reppsitory's visibility and final click on "Create repository".
> > - Pass the link to the repo to your colleagues, who will be the contributors.

Reviewer: Just some re-wording for clarity:
```suggestion
> - Set the owner of the new repository to your own GitHub username.
> - Set the name of the repository to "book_of_recipes".
> - Set the description to something like "Repository for the exercises of the Grad School Git Course".
> - Make sure to select the "Public" option for your repsitory's visibility.
> - Click on "Create repository".
> - Pass the link for your repo to the other group members, who will be the contributors.
```

> > - Pass the link to the repo to your colleagues (who will be the contributors).

Reviewer 2: ```suggestion
> - Pass the link to the repo to your colleagues, who will be the contributors.
```

>
> Now, start collaborating!
>

> > **Create a new release** (Administrator's task)

Reviewer: For consistency with the below headings

```suggestion
> **Create a new release** (Administrator's task):
```

>
> - Create a new release, let's say `1.0.0`, as the starting point of the book.
>
> **Creating new issues** (to be done by everyone):
>
> - Open new issues with recipes for sauces you would like to have in the book.
>
> **Explore existing issues** (Contributor's task):
>

> > - Comment on the issues that you want to be assigned to. Remember, you neither have write access to the repository nor you are not part of the same organisation of the repository (in this case the repository belongs to the Administrator is not a part of an organisation), the only way of being assigned to an issue is by making a comment on the issue.

Reviewer 2: I think this could be terser. How about:

```suggestion
> - Comment on the issues that you want to be assigned to. (As you don't have write access to the repository, you cannot assign yourself to an issue directly.)
```

>
> **Task assignment** (Administrator's task):
>
> - Add some tags, prioritise some of the recipes, and assign yourself or one of your colleagues as responsible for each of them.
>
> **Working on a branch, pushing the changes, and opening a PR** (Contributor's tasks):
>

> > - Fork the Administrator's repository and clone it locally.

Reviewer: I'd be inclined to but this forking task at the top, right after the Housekeeping section. Just so they can get set up while the administrator creates the release

> > - Create a branch and work on the recipes you have been assigned. Practice the concepts learnt in previous episodes about making the changes locally and pushing those changes back to the remote repository. You can even try a [gitflow approach](https://nvie.com/posts/a-successful-git-branching-model/) if you feel ambitious!

Reviewer 2: I know you didn't add this bit, but I never liked that we mention the gitflow approach here, which we don't discuss anywhere else. Plus, it's not a particularly good way to work these days anyway (e.g. we don't do it!).

```suggestion
> - Create a branch and work on the recipes you have been assigned. Practice the concepts learnt in previous episodes about making the changes locally and pushing those changes back to the remote repository.
```

> - When ready, open a PR to the Administrator's repo and request their review.
>
> **Reviewing and Merging PRs** (Administrator's tasks):
>
> - Review the PR, request some changes and accept others. Make sure the relevant checks performed by the continuous integration system are all passing.
> - When ready, merge the PR.
>
> **Creating a new release** (Administrator's task):
>

> > - When all the recipes by the Contributor's are added, create a new release.

Reviewer: ```suggestion
> - When all the recipes by the Contributors are added, create a new release.
```

>
> These exercises can be repeated with the other members of the group acting

> > now as administrators and choosing a different topic for the recipes (eg

Reviewer: I would remove this footnote

> > - Whenever there is a new recipe added, create a new release.

Reviewer 2: Do we want them to be making so many releases? They make one release at the start -- would it be enough to have them just make another release at the end?

Reviewer: Probably one at the start and another one later is enough

Author: I can see what you mean
